components/header.js

```diff
@@ -11,6 +43,106 @@ const Header = () => {
           <li><button>Dark Mode</button></li>
         </ul>
       </nav>
+      <div className={pageStyles.rules}>
+        <div className={pageStyles.cardRanks}>
+          <span>
+            <h2>Card Ranks (High - Low)</h2>
+            <button className={headerRules.cardRank ? `${pageStyles.buttonCaret} ${pageStyles.divOpen}` : pageStyles.buttonCaret} onClick={handleCardRankClick}>
+              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"/></svg>
+            </button>
+          </span>
+          <div className={headerRules.cardRank ? pageStyles.listCardRank : pageStyles.hide}>2 A K Q J 10 9 8 7 6 5 4 3 2 1</div>
+        </div>
+        <div className={pageStyles.suitRanks}>
+          <span>
+            <h2>Suit Ranks (High - Low)</h2>
+            <button className={headerRules.suitRank ? `${pageStyles.buttonCaret} ${pageStyles.divOpen}` : pageStyles.buttonCaret} onClick={handleSuitRankClick}>
+              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"/></svg>
+            </button>
+          </span>
+          <div className={headerRules.suitRank ? pageStyles.listCardRank : pageStyles.hide}>Hearts - Diamonds - Clubs - Spades</div>
+        </div>
+        <div className={pageStyles.ruleCombinations}>
+          <span>
+            <h2>Valid Combinations</h2>
+            <button className={headerRules.combinations ? `${pageStyles.buttonCaret} ${pageStyles.divOpen}` : pageStyles.buttonCaret} onClick={handleCombinationsClick}>
+              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"/></svg>
+            </button>
+          </span>
```

Comment:
a lot of these are repetitive as well. I would look into turning these into a component and just passing in the props that make them differ to DRY up your code.